cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -797,3 +805,136 @@ func installedPkgDetailFromRelease(r *release.Release, ref *corev1.InstalledPack
 		CustomDetail: customDetailHelm,
 	}, nil
 }
+
+func splitChartIdentifier(chartID string) (repoName, chartName string, err error) {
+	// getUnescapedChartID also ensures that there are two parts (ie. repo/chart-name only)
+	unescapedChartID, err := getUnescapedChartID(chartID)
+	if err != nil {
+		return "", "", err
+	}
+	chartIDParts := strings.Split(unescapedChartID, "/")
+	return chartIDParts[0], chartIDParts[1], nil
+}
+
+// CreateInstalledPackage creates an installed package.
+func (s *Server) CreateInstalledPackage(ctx context.Context, request *corev1.CreateInstalledPackageRequest) (*corev1.CreateInstalledPackageResponse, error) {
+	// Get the AppRepository for the available package.
+	// TODO: currently app repositories are only supported on the cluster on
+	// which Kubeapps is installed. #1982
+	chartID := request.GetAvailablePackageRef().GetIdentifier()
+	repoNamespace := request.GetAvailablePackageRef().GetContext().GetNamespace()
+	repoName, chartName, err := splitChartIdentifier(chartID)
+	chartVersion := request.GetPkgVersionReference().GetVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	// Most of the existing code that we want to reuse is based on having a typed AppRepository.
+	appRepo, caCertSecret, authSecret, err := s.getAppRepoAndRelatedSecrets(ctx, repoName, repoNamespace)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable to fetch app repo %q from namespace %q: %v", repoName, repoNamespace, err)
+	}
+
+	// Grab the chart itself
+	ch, err := handlerutil.GetChart(
+		&chart.Details{
+			AppRepositoryResourceName:      appRepo.Name,
+			AppRepositoryResourceNamespace: appRepo.Namespace,
+			ChartName:                      chartName,
+			Version:                        chartVersion,
+		},
+		appRepo,
+		caCertSecret, authSecret,
+		// TODO(minelson): add a useragent comment to kubeapps APIs and ensure it is passed
+		// through to be used here.
+		s.chartClientFactory.New(appRepo.Spec.Type, "kubeapps-apis/devel"),
```

Comment:
Weren't we already using the proper user-agent?

Reply:
You reviewed that branch first, but it was done after this one and targeted this branch (when you reviewed it... I'll be rebasing it after landing this one).